frontend/src/App.jsx

```diff
@@ -135,20 +149,20 @@ function App() {
     setActiveModal("");
   };
 
-  const handleLogin = (credentials) => {
-    return loginUser(credentials)
+  const handleLogin = (credentials) =>
+    loginUser(credentials)
       .then((res) => {
         if (res.token) {
           localStorage.setItem("jwt", res.token);
           setIsLoggedIn(true);
-          checkToken(res.token).then((res) => {
-            setUser(res.data);
+          checkToken(res.token).then((res2) => {
+            console.log(res2);
+            setUser(res2.data);
             setActiveModal("");
```

Comment:
`setActiveModal("")`


Please, reuse the `closeActiveModal` function for any closing, to avoid code duplication